server/mediaserver.go

```diff
@@ -540,6 +540,7 @@ func (s *LivepeerServer) HandlePush(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	r.Body.Close()
+	r.URL = &url.URL{Scheme: "http", Host: r.Host, Path: r.URL.Path}
```

Comment:
Naturally we won't always have plain `http` but this is probably OK for now. Just double checking, does the `r.Host` field actually contain the hostname data we need?

Reply:
It does indeed! I tested it locally under a number of different scenarios and it works